My Series List/My Series List/View/ListView/ListPresenter.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Presenter.swift
+//  My Series List
+//
+//  Created by Ivan Dias on 24/11/2021.
+//
+
+import Foundation
+import UIKit
+
+protocol ListView: NSObjectProtocol {
+    
+    func startLoading()
+    func finishLoading()
+    func listSeries(_ series: [Series])
+    func listError(_ error: Error)
+    func listEmpty()
+}
+
+class ListPresenter: NSObject {
+    
+    var listView: ListView?
+    let seriesRepository = SeriesRepository()
+    var isChecked = true
+    var series: [Series] = []
+    
+    func getSeries() {
+        seriesRepository.getSeries(completionHandler: { coreDataSeries in
+            switch coreDataSeries {
+            case .failure(let error):
+                self.listView?.listError(error)
+            case .success(let coreDataSeries):
+                self.series = coreDataSeries
+                if self.series.isEmpty {
+                    self.listView?.listEmpty()
```

Comment:
I believe either listEmpty or listSeries should be called. Please review success case

Reply:
I don't understand quite well what you mean in this case, I call both methods inside the success case scope

Reply:
That's the problem. You should just call one or the other. If your list is empty listEmpty() and listSeries() will both be called. Try to use an if else to just call one of the methods